frontend/src/metabase/query_builder/components/chart-type-selector/ChartTypeOption/ChartTypeOption.unit.spec.tsx

```diff
@@ -0,0 +1,134 @@
+import userEvent from "@testing-library/user-event";
+
+import {
+  getIcon,
+  queryIcon,
+  renderWithProviders,
+  screen,
+} from "__support__/ui";
+import type { IconName } from "metabase/ui";
+import registerVisualizations from "metabase/visualizations/register";
+import type { CardDisplayType } from "metabase-types/api";
+
+import { ChartTypeOption, type ChartTypeOptionProps } from "./ChartTypeOption";
+
+registerVisualizations();
+
+const EXPECTED_VISUALIZATION_VALUES: Record<
+  CardDisplayType,
+  {
+    key: CardDisplayType;
+    displayName: string;
+    iconName: IconName | (string & unknown);
+  }
+> = {
+  scalar: { key: "scalar", displayName: "Number", iconName: "number" },
+  smartscalar: {
+    key: "smartscalar",
+    displayName: "Trend",
+    iconName: "smartscalar",
+  },
+  progress: { key: "progress", displayName: "Progress", iconName: "progress" },
+  gauge: { key: "gauge", displayName: "Gauge", iconName: "gauge" },
+  table: { key: "table", displayName: "Table", iconName: "table2" },
+  line: { key: "line", displayName: "Line", iconName: "line" },
+  area: { key: "area", displayName: "Area", iconName: "area" },
+  bar: { key: "bar", displayName: "Bar", iconName: "bar" },
+  waterfall: {
+    key: "waterfall",
+    displayName: "Waterfall",
+    iconName: "waterfall",
+  },
+  combo: { key: "combo", displayName: "Combo", iconName: "lineandbar" },
+  row: { key: "row", displayName: "Row", iconName: "horizontal_bar" },
+  scatter: { key: "scatter", displayName: "Scatter", iconName: "bubble" },
+  pie: { key: "pie", displayName: "Pie", iconName: "pie" },
+  map: { key: "map", displayName: "Map", iconName: "pinmap" },
+  funnel: { key: "funnel", displayName: "Funnel", iconName: "funnel" },
+  object: { key: "object", displayName: "Detail", iconName: "document" },
+  pivot: { key: "pivot", displayName: "Pivot Table", iconName: "pivot_table" },
+};
+
+const setup = ({
+  selectedVisualization = "table",
+  onSelectVisualization = jest.fn(),
+  visualizationType = "bar",
+}: Partial<ChartTypeOptionProps> = {}) => {
+  renderWithProviders(
+    <ChartTypeOption
+      selectedVisualization={selectedVisualization}
+      onSelectVisualization={onSelectVisualization}
+      visualizationType={visualizationType}
+    />,
+  );
+
+  return { onSelectVisualization };
+};
+
+describe("ChartTypeOption", () => {
+  Object.entries(EXPECTED_VISUALIZATION_VALUES).forEach(
```

Comment:
`EXPECTED_VISUALIZATION_VALUES` is an object where key is also in the value. Wdyt about making it an array, and then using `it.each(EXPECTED_VISUALIZATION_VALUES)`? It would help to remove type casts

Reply:
yes!